sdk/src/main/java/io/dapr/client/DaprClientGrpc.java

```diff
@@ -185,6 +192,78 @@ public Mono<Void> publishEvent(PublishEventRequest request) {
     }
   }
 
+  /**
+   *
+   * {@inheritDoc}
+   */
+  public <T> Mono<BulkPublishResponse> publishEvents(BulkPublishRequest<T> request) {
+    try {
+      String pubsubName = request.getPubsubName();
+      String topic = request.getTopic();
+      DaprProtos.BulkPublishRequest.Builder envelopeBuilder = DaprProtos.BulkPublishRequest.newBuilder();
+      envelopeBuilder.setTopic(topic);
+      envelopeBuilder.setPubsubName(pubsubName);
+
+      for (BulkPublishRequestEntry<?> entry: request.getEntries()) {
+        Object event = entry.getEvent();
+        byte[] data;
+        String contentType = entry.getContentType();
+
+        // Serialize event into bytes
+        if (!Strings.isNullOrEmpty(contentType) && objectSerializer instanceof DefaultObjectSerializer) {
+          // If content type is given by user and default object serializer is used
+          data = DefaultContentTypeConverter.convertEventToBytesForGrpc(event, contentType);
+        } else {
+          // perform the serialization as per user given input of serializer
+          // this is also the case when content type is empty
+          data = objectSerializer.serialize(event);
+          if (Strings.isNullOrEmpty(contentType)) {
+            // Only override content type if not given in input by user
+            contentType = objectSerializer.getContentType();
+          }
+        }
+
+        DaprProtos.BulkPublishRequestEntry.Builder reqEntryBuilder = DaprProtos.BulkPublishRequestEntry.newBuilder()
+            .setEntryId(entry.getEntryId())
+            .setEvent(ByteString.copyFrom(data))
+            .setContentType(contentType);
+        Map<String, String> metadata = entry.getMetadata();
+        if (metadata != null) {
+          reqEntryBuilder.putAllMetadata(metadata);
+        }
+        envelopeBuilder.addEntries(reqEntryBuilder.build());
+      }
+
+      // Set metadata if available
+      Map<String, String> metadata = request.getMetadata();
+      if (metadata != null) {
+        envelopeBuilder.putAllMetadata(metadata);
+      }
+
+      return Mono.subscriberContext().flatMap(
+          context ->
+              this.<DaprProtos.BulkPublishResponse>createMono(
+                  it -> intercept(context, asyncStub).bulkPublishEventAlpha1(envelopeBuilder.build(), it)
+              )
+      ).map(
+          it -> {
+            BulkPublishResponse response = new BulkPublishResponse();
+            List<BulkPublishResponseFailedEntry> entries = new ArrayList<>();
```

Comment:
nit: Shouldn't we name it as `failedEntries` instead of `entries` since it explicitly contains only the failed entries ?